packages/features/eventtypes/components/EventType.tsx

```diff
@@ -471,6 +490,46 @@ export const EventType = (props: EventTypeSetupProps & { allActiveWorkflows?: Wo
     return updatedFields;
   };
 
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  const getEachFieldState = (parentKey: string, objValue): boolean => {
+    let isDirty = false;
+    isDirty = getFieldState(parentKey as keyof FormValues).isDirty;
+    // if field of type array (or object) itself has dirty state.
+    if (isDirty) {
+      return isDirty;
+    }
+    for (const key in objValue) {
+      const completeKey = `${parentKey}.${key}`;
+      if (isObject(objValue[key]) || isArray(objValue[key])) {
+        // Recursive call for nested objects or arrays within each element
+        isDirty = getEachFieldState(completeKey, objValue[key]);
+      } else {
+        isDirty = getFieldState(completeKey as keyof FormValues).isDirty;
+      }
+      if (isDirty) {
+        return isDirty;
+      }
+    }
+    return isDirty;
+  };
+
+  const getFormFieldStates = (values: FormValues): boolean => {
+    let isDirty = false;
+    for (const key in values) {
+      const typedKey = key as keyof typeof values;
+      if (isObject(values[typedKey]) || isArray(values[typedKey])) {
+        isDirty = getEachFieldState(typedKey, values[typedKey]);
+      } else {
+        isDirty = getFieldState(typedKey).isDirty;
+      }
+      if (isDirty) {
+        return isDirty;
+      }
+    }
+    return isDirty;
+  };
```

Comment:
isFormDirty will be active only when form is submitted.
But here, we want to know the form state or each field state without submitting the Form, Hence `getFieldState` is used on each of the field in `FormValues`.
`getEachFieldState()` is created to run recursively to handle nested objects and arrays.

Reply:
These functions are implemented such a way that -
If any one of field has `Dirty` state, then no need to check for remaining fields.

Reply:
We already have `getDirtyFields()`, but we can't use it for Unsaved Changes scenario, because -
it ultimates makes use of `dirtyFields` array returned from `formMethods`, which will be populated or activated only when form is submitted.